python/pyarrow/tests/test_sparse_tensor.py

```diff
@@ -232,11 +255,65 @@ def test_dense_to_sparse_tensor(dtype_str, arrow_type, sparse_tensor_type):
                       [0, 7, 0, 0],
                       [0, 0, 0, 0],
                       [0, 0, 0, 5]]).astype(dtype)
+    dim_names = ('x', 'y')
 
-    sparse_tensor = sparse_tensor_type.from_dense_numpy(array)
+    sparse_tensor = sparse_tensor_type.from_dense_numpy(array, dim_names)
     tensor = sparse_tensor.to_tensor()
     result_array = tensor.to_numpy()
 
     assert sparse_tensor.type == arrow_type
     assert tensor.type == arrow_type
+    assert sparse_tensor.dim_names == dim_names
     assert np.array_equal(array, result_array)
+
+
+@pytest.mark.skipif(not coo_matrix, reason="requires scipy")
+@pytest.mark.parametrize('dtype_str,arrow_type', tensor_type_pairs)
+def test_sparse_coo_tensor_scipy_roundtrip(dtype_str, arrow_type):
+    dtype = np.dtype(dtype_str)
+    data = np.array([1, 2, 3, 4, 5, 6]).astype(dtype)
+    row = np.array([0, 0, 2, 3, 1, 3])
+    col = np.array([0, 2, 0, 4, 5, 5])
+    shape = (4, 6)
+    dim_names = ('x', 'y')
+
+    sparse_array = coo_matrix((data, (row, col)), shape=shape)
+    sparse_tensor = pa.SparseCOOTensor.from_scipy(sparse_array,
+                                                  dim_names=dim_names)
+    out_sparse_array = sparse_tensor.to_scipy()
+
+    assert sparse_tensor.type == arrow_type
+    assert sparse_tensor.dim_names == dim_names
+    assert sparse_array.dtype == out_sparse_array.dtype
+    assert np.array_equal(sparse_array.data, out_sparse_array.data)
+    assert np.array_equal(sparse_array.row, out_sparse_array.row)
+    assert np.array_equal(sparse_array.col, out_sparse_array.col)
+    if dtype_str != 'f2':
```

Comment:
I seems scipy.sparse doesn't support `sparse_coo_array.toarray()` with 'f2' type.

Reply:
There's a an [open issue](https://github.com/scipy/scipy/issues/7408) for this. Using the [suggested workaround](https://github.com/scipy/scipy/issues/7408#issuecomment-537804284) for now.